src/libraries/System.Private.Uri/src/System/UriHelper.cs

```diff
@@ -570,51 +570,58 @@ internal static bool IsLWS(char ch)
             return (ch <= ' ') && (ch == ' ' || ch == '\n' || ch == '\r' || ch == '\t');
         }
 
-        //
+        private const char BidiControlCharacterMin = '\u200E';
+        private const char BidiControlCharacterMax = '\u202E';
+
         // Is this a Bidirectional control char.. These get stripped
-        //
         internal static bool IsBidiControlCharacter(char ch)
         {
-            return (ch == '\u200E' /*LRM*/ || ch == '\u200F' /*RLM*/ || ch == '\u202A' /*LRE*/ ||
-                    ch == '\u202B' /*RLE*/ || ch == '\u202C' /*PDF*/ || ch == '\u202D' /*LRO*/ ||
-                    ch == '\u202E' /*RLO*/);
+            return char.IsBetween(ch, BidiControlCharacterMin, BidiControlCharacterMax)
+                && IsBidiControlCharacterSlow(ch);
+
+            static bool IsBidiControlCharacterSlow(char ch)
+            {
+                return (ch == '\u200E' /*LRM*/ || ch == '\u200F' /*RLM*/ || ch == '\u202A' /*LRE*/ ||
+                        ch == '\u202B' /*RLE*/ || ch == '\u202C' /*PDF*/ || ch == '\u202D' /*LRO*/ ||
+                        ch == '\u202E' /*RLO*/);
+            }
         }
 
-        //
         // Strip Bidirectional control characters from this string
-        //
         internal static unsafe string StripBidiControlCharacters(ReadOnlySpan<char> strToClean, string? backingString = null)
         {
             Debug.Assert(backingString is null || strToClean.Length == backingString.Length);
 
             int charsToRemove = 0;
-            foreach (char c in strToClean)
+
+            int indexOfPossibleCharToRemove = strToClean.IndexOfAnyInRange(BidiControlCharacterMin, BidiControlCharacterMax);
+            if (indexOfPossibleCharToRemove >= 0)
             {
-                if ((uint)(c - '\u200E') <= ('\u202E' - '\u200E') && IsBidiControlCharacter(c))
+                // Slow path: Contains chars that fall in the [u200E, u202E] range (so likely Bidi)
+                foreach (char c in strToClean.Slice(indexOfPossibleCharToRemove))
                 {
-                    charsToRemove++;
+                    if (IsBidiControlCharacter(c))
+                    {
+                        charsToRemove++;
+                    }
                 }
             }
 
             if (charsToRemove == 0)
             {
+                // Hot path
                 return backingString ?? new string(strToClean);
             }
 
-            if (charsToRemove == strToClean.Length)
-            {
-                return string.Empty;
-            }
```

Comment:
Removing this because it's really unlikely and string.Create will handle the 0 case?

Reply:
Yes